src/test/java/teammates/test/cases/common/UrlTest.java

```diff
@@ -16,4 +18,122 @@ public void testTrimTrailingSlash(){
+            ignoreExpectedException();
+        }
+        
+    }
+    
+    @Test
+    public void testGetParameter() {
+        Url url = new Url("http://www.google.com/page?key1=value1&key2=value2&key3=value3");
+        assertEquals("value1", url.get("key1"));
+        assertEquals("value2", url.get("key2"));
+        assertEquals("value3", url.get("key3"));
+        assertEquals(null, url.get("key4"));
+    }
+    
+    @Test
+    public void testAppendParameters() {
+        
+        ______TS("static method addParamToUrl");
+        
+        String url = "http://www.google.com";
+        assertEquals(url, Url.addParamToUrl(url, null, "value"));
+        assertEquals(url, Url.addParamToUrl(url, "", "value"));
+        assertEquals(url, Url.addParamToUrl(url, "key", null));
+        assertEquals(url, Url.addParamToUrl(url, "key", ""));
+        assertEquals(url + "?key1=value1", Url.addParamToUrl(url, "key1", "value1"));
+        url += "?key1=value1";
+        assertEquals(url + "&key2=value2", Url.addParamToUrl(url, "key2", "value2"));
+        url += "&key2=value2";
+        assertEquals(url, Url.addParamToUrl(url, "key1", "newvalue1"));
+        assertEquals(url, Url.addParamToUrl(url, "key2", "newvalue2"));
+        assertEquals(url + "&key3=" + Sanitizer.sanitizeForUri("#& ?"), Url.addParamToUrl(url, "key3", "#& ?"));
+        
+        ______TS("in-place method withParam");
+        
+        Url newUrl = new Url("http://www.google.com/page");
+        newUrl.withParam("key1", "value1");
+        assertEquals("/page?key1=value1", newUrl.toString());
+        newUrl.withParam("key1", "newvalue1");
+        assertEquals("http://www.google.com/page?key1=value1", newUrl.toAbsoluteString());
+    }
+    
+    @Test
+    public void testAppUrlAssertion() {
+        
+        ______TS("typical non-empty case");
+        
+        AppUrl url = new AppUrl("http://www.google.com/page");
+        assertEquals("/page", url.toString());
+        assertEquals("http://www.google.com/page", url.toAbsoluteString());
+        
+        ______TS("empty path case");
+        
+        url = new AppUrl("http://www.google.com");
+        assertEquals("", url.toString());
+        assertEquals("http://www.google.com", url.toAbsoluteString());
+        
+        ______TS("malformed URL: not http(s)");
+        
+        try {
+            url = new AppUrl("file:///C:/path/to/file.ext");
+            signalFailureToDetectException();
+        } catch (AssertionError ae) {
+            ignoreExpectedException();
+        }
+        
+    }
+    
+    @Test
+    public void testGetRelativePath() {
+        
+        ______TS("web URL with no relative path");
+        
+        String url = "http://www.google.com";
+        assertEquals("", Url.getRelativePath(url));
+        
+        ______TS("typical web URL");
+        
+        url = "http://www.google.com/page";
+        assertEquals("/page", Url.getRelativePath(url));
+        
+    }
```

Comment:
If it's about how the value from `*.properties` gets trimmed, that wouldn't be tested from `*Url` class, but `*PropertiesTest` instead. Otherwise there's no such thing as "trimming"; `new Url("http://www.google.com/")` will be interpreted as `http://www.google.com` as base URL, and `/` as relative path.